src/odemis/gui/cont/correlation.py

```diff
@@ -0,0 +1,242 @@
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+import logging
+import numpy
+import wx
+# IMPORTANT: wx.html needs to be imported for the HTMLWindow defined in the XRC
+# file to be correctly identified. See: http://trac.wxwidgets.org/ticket/3626
+# This is not related to any particular wxPython version and is most likely permanent.
+import wx.html
+from odemis import model
+import odemis.acq.stream as acqstream
+import odemis.gui.model as guimod
+from odemis.acq.stream import StaticStream
+
+# TODO: move to more approprate location
+def updateImageInViews(s: StaticStream, views: list):
+    """force update the static stream in the selected views
+    param: s (StaticStream) the static stream to update
+    param: views (list[View]) the list of views to update"""
+    v: guimod.View 
+    for v in views:
+        for sp in v.stream_tree.getProjections():  # stream or projection
+            if isinstance(sp, acqstream.DataProjection):
+                st = sp.stream
+            else:
+                st = sp
+            if st is s:
+                sp._shouldUpdateImage()
+
+class CorrelationController(object):
+
+    def __init__(self, tab_data, panel, tab, viewports):
+        """
+        tab_data (MicroscopyGUIData): the representation of the microscope GUI
+        panel (wx._windows.Panel): the panel containing the UI controls
+        tab: (Tab): the tab which should show the data
+        viewports: (ViewPorts): the tab view ports
+        """
+        
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._panel = panel
+        self._tab = tab
+        self._viewports = viewports
+
+        # the correlation streams, and selected stream
+        self.correlationStreams = model.ListVA()
+        self.selected_stream = model.VigilantAttribute(None)
+
+        # disable if no streams are present
+        self.correlationStreams.subscribe(self._on_correlation_streams_change, init=True)
+
+        # connect the correlation streams to the tab data
+        self._panel.cmb_correlation_stream.Bind(wx.EVT_COMBOBOX, self._on_selected_stream_change)
+
+        for vp in self._viewports:
+            vp.canvas.Bind(wx.EVT_CHAR, self._on_char)
+            vp.canvas.Bind(wx.EVT_LEFT_DOWN, self._on_mouse_down)
+       
+        # TODO: add toggle for enablling moving the streams. 
+        # If not enabled, the streams dont move, use standard controls for mouse / keyboard events...
+        # TODO: add reset button to reset the stream position to the original position
+
+    def _on_correlation_streams_change(self, streams):
```

Comment:
Typically, every VA listener which calls a wxPython function should be protected with `@call_in_wx_main()`. Otherwise it may sometimes crash the whole GUI. (because the VA might be changed from any thread, not necessarily the main GUI thread)